### ISSUE TITLE
Vscode Jupyter Python Docstring Viewer Doens't Work Properly

### ISSUE THREAD

**Issue by rcgalbo** [microsoft/vscode-jupyter#14992]
Type: <b>Bug</b>

I am using vscode to edit jupyter notebooks. I have a local python library that is installed with pip. In a vanilla text document, when I hover over the function name I am able to view the docstring. When I hover over the function name in the jupyter notebook I see "(import) <function_name>: Any" I would really like to be able to read the docstring from the jupyter notebook. Any suggestions would be very appreciated.

VS Code version: Code - Insiders 1.86.0-insider (d8a75eacf7c70e7327a460a8224dec14a65e1e27, 2024-01-04T05:36:53.531Z)
OS version: Windows_NT x64 10.0.22624
Modes:
Remote OS version: Linux x64 5.15.133.1-microsoft-standard-WSL2

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|12th Gen Intel(R) Core(TM) i7-12700H (20 x 2688)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.68GB (14.86GB free)|
|Process Argv|--crash-reporter-id f694ba95-d980-46d2-b4e7-4c3d2cd0a0a5|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu|
|OS|Linux x64 5.15.133.1-microsoft-standard-WSL2|
|CPUs|12th Gen Intel(R) Core(TM) i7-12700H (20 x 2688)|
|Memory (System)|15.46GB (11.84GB free)|
|VM|0%|
</details><details><summary>Extensions (11)</summary>

Extension|Author (truncated)|Version
---|---|---
jupyter-keymap|ms-|1.1.2
remote-wsl|ms-|0.81.8
black-formatter|ms-|2023.7.13471009
python|ms-|2023.23.13541005
vscode-pylance|ms-|2023.12.1
jupyter|ms-|2024.1.2024010501
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.17
tensorboard|ms-|2023.10.1002992421
vscode-jupyter-cell-tags|ms-|0.1.8
vscode-jupyter-slideshow|ms-|0.1.5

</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vshan820:30294714
vscod805:30301674
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
py29gd2263:30784851
vscaat:30438846
vsclangdf:30492506
c4g48928:30535728
pythongtdpath:30726887
i26e3531:30792625
welcomedialog:30812478
pythonidxpt:30768918
pythonnoceb:30776497
asynctok:30898717
dsvsc013:30777762
dsvsc014:30777825
dsvsc015:30821418
pythontestfixt:30866404
pythonregdiag2:30926734
pyreplss1:30879911
pythonmypyd1:30859725
pythoncet0:30859736
pythontbext0:30879054
accentitlementst:30870582
dsvsc016:30879898
dsvsc017:30880771
dsvsc018:30880772
aa_t_chat:30882232
cp7184t3:30927821

```

</details>

**Comment by DonJayamanne** [comment]
Thanks for filing this issue and sorry you are running into this issue.
Please can you provide the following information:
* Python environment used in the following scenario
>  In a vanilla text document, when I hover over the function name I am able to view the docstring. 

* The kernel selected in the Jupyter notebook in the followin scenario
> hen I hover over the function name in the jupyter notebook I see "(import) <function_name>: Any" 

Basically the python env selected in your workspace folder (using Python extension) must match the same environment selected as a Jupyter kernel in the Jupyter notebook (this is done via the Kernle picker on the top right hard corner of the notebook)

**Comment by rcgalbo** [comment]
The issue I am having is with a locally installed package that I have developed. I am wondering if the issue is somehow related to how I created the package. I have been unable to figure this out for a few weeks now. Any assistance would be greatly appreciated.

**Python Version:** Python 3.10.12

**Kernel:**
![image](https://github.com/microsoft/vscode-jupyter/assets/6351996/9e66db22-de5a-4dea-9eae-25d28fc70f65)

**Packages installed:**
Package                      Version      Editable project location
---------------------------- ------------ ----------------------------
absl-py                      2.0.0
aiohttp                      3.9.0
aiosignal                    1.3.1
alembic                      1.12.1
alpaca-py                    0.10.0
and                          66.0.2
anyio                        3.7.1
asttokens                    2.3.0
astunparse                   1.6.3
async-timeout                4.0.3
attrs                        23.1.0
backcall                     0.2.0
cachetools                   5.3.2
certifi                      2023.7.22
chardet                      3.0.4
charset-normalizer           3.2.0
click                        8.1.7
cloudpickle                  3.0.0
cmake                        3.27.7
cmdstanpy                    1.2.0
colorlog                     6.7.0
comm                         0.1.4
contourpy                    1.1.0
cramjam                      2.7.0
cycler                       0.11.0
dask                         2023.12.0
debugpy                      1.6.7.post1
decorator                    5.1.1
distributed                  2023.12.0
dm-tree                      0.1.8
eventkit                     1.0.1
exceptiongroup               1.1.3
executing                    1.2.0
fake-useragent               1.2.1
fastapi                      0.104.1
fastjsonschema               2.18.0
fastparquet                  2023.10.1
filelock                     3.13.1
flatbuffers                  23.5.26
fonttools                    4.42.1
frozenlist                   1.4.0
fsspec                       2023.10.0
futures                      1.0.0        /home/rcgalbo/stonks/futures
gast                         0.4.0
google-auth                  2.23.4
google-auth-oauthlib         1.0.0
google-pasta                 0.2.0
graphviz                     0.20.1
greenlet                     3.0.1
grpcio                       1.59.3
h5py                         3.10.0
holidays                     0.36
ib-insync                    0.9.86
ibapi                        10.25.1
idna                         2.10
importlib-metadata           7.0.0
importlib-resources          6.1.1
inflection                   0.5.1
ipykernel                    6.25.1
ipython                      8.15.0
jax                          0.4.21
jedi                         0.19.0
Jinja2                       3.1.2
joblib                       1.3.2
jsonschema                   4.19.0
jsonschema-specifications    2023.7.1
jupyter_client               8.3.1
jupyter_core                 5.3.1
keras                        2.15.0
kiwisolver                   1.4.5
libclang                     16.0.6
lightning                    2.1.2
lightning-utilities          0.10.0
lit                          17.0.5
llvmlite                     0.41.1
locket                       1.0.0
lxml                         4.9.3
Mako                         1.3.0
Markdown                     3.5.1
markdown-it-py               3.0.0
MarkupSafe                   2.1.3
matplotlib                   3.7.2
matplotlib-inline            0.1.6
mdurl                        0.1.2
ml-dtypes                    0.2.0
more-itertools               10.1.0
mpmath                       1.3.0
msgpack                      1.0.5
multidict                    6.0.4
namex                        0.0.7
Nasdaq-Data-Link             1.0.4
nbformat                     5.9.2
nest-asyncio                 1.5.7
networkx                     3.2.1
numba                        0.58.1
numpy                        1.23.5
nvidia-cublas-cu12           12.2.5.6
nvidia-cuda-cupti-cu12       12.2.142
nvidia-cuda-nvcc-cu12        12.2.140
nvidia-cuda-nvrtc-cu12       12.2.140
nvidia-cuda-runtime-cu12     12.2.140
nvidia-cudnn-cu12            8.9.4.25
nvidia-cufft-cu12            11.0.8.103
nvidia-curand-cu12           10.3.3.141
nvidia-cusolver-cu12         11.5.2.141
nvidia-cusparse-cu12         12.1.2.141
nvidia-nccl-cu12             2.16.5
nvidia-nvjitlink-cu12        12.2.140
nvidia-nvtx-cu12             12.1.105
oauthlib                     3.2.2
opt-einsum                   3.3.0
optuna                       3.4.0
packaging                    23.1
pandas                       2.1.0
parso                        0.8.3
partd                        1.4.1
patsy                        0.5.3
pexpect                      4.8.0
pickleshare                  0.7.5
Pillow                       10.0.0
pip                          22.0.2
platformdirs                 3.10.0
plotly                       5.16.1
prompt-toolkit               3.0.39
prophet                      1.1.5
protobuf                     4.23.4
psutil                       5.9.5
ptyprocess                   0.7.0
pure-eval                    0.2.2
pyarrow                      14.0.1
pyasn1                       0.5.1
pyasn1-modules               0.3.0
pydantic                     1.10.12
pydot                        1.4.2
Pygments                     2.16.1
pyparsing                    3.0.9
python-dateutil              2.8.2
pytimeparse                  1.1.8
pytorch-forecasting          1.0.0
pytorch-lightning            2.1.2
pytorch_optimizer            2.12.0
pytorch-triton-rocm          2.1.0
pytz                         2023.3
PyYAML                       6.0.1
pyzmq                        25.1.1
Quandl                       3.7.0
referencing                  0.30.2
requests                     2.31.0
requests-oauthlib            1.3.1
rich                         13.7.0
rpds-py                      0.10.0
rsa                          4.9
scikit-learn                 1.3.2
scipy                        1.11.3
seaborn                      0.13.0
setuptools                   59.6.0
six                          1.16.0
sniffio                      1.3.0
sodapy                       2.2.0
sortedcontainers             2.4.0
SQLAlchemy                   2.0.23
sseclient-py                 1.8.0
stack-data                   0.6.2
stanio                       0.3.0
starlette                    0.27.0
statsmodels                  0.14.0
stumpy                       1.12.0
sympy                        1.12
tblib                        3.0.0
tenacity                     8.2.3
tensorboard                  2.15.1
tensorboard-data-server      0.7.2
tensorflow                   2.15.0.post1
tensorflow-estimator         2.15.0
tensorflow-io-gcs-filesystem 0.34.0
termcolor                    2.4.0
threadpoolctl                3.2.0
toolz                        0.12.0
torch                        2.1.1
torch-tb-profiler            0.4.3
torchmetrics                 1.2.0
tornado                      6.3.3
tqdm                         4.66.1
traitlets                    5.9.0
triton                       2.1.0
tsfresh                      0.20.1
twelvedata                   1.2.12
typing_extensions            4.8.0
tzdata                       2023.3
urllib3                      1.25.11
wcwidth                      0.2.6
websockets                   10.4
Werkzeug                     3.0.1
wheel                        0.42.0
wrapt                        1.14.1
yarl                         1.9.3
zict                         3.0.0
zipp                         3.17.0

**Comment by DonJayamanne** [comment]
What is  Python environment used in the following scenario
> In a vanilla text document, when I hover over the function name I am able to view the docstring.

**Comment by rcgalbo** [comment]
The same python environment as listed above.

**Comment by DonJayamanne** [comment]
> The issue I am having is with a locally installed package that I have developed.

what is the name and where is this installed
By where, I mean what folder? Is it site packages folder of stink-env?

**Comment by rcgalbo** [comment]
![image](https://github.com/microsoft/vscode-jupyter/assets/6351996/6920a8ac-7be9-48a7-bb3f-24a6ff793e0f)
The environment is installed in the project. It is a virtual environment.

**Comment by DonJayamanne** [comment]
What’s the name of the package and where is it installed?

**Comment by rcgalbo** [comment]
The package is a local package called `futures` and is installed locally in the python environment `stonks-env`

**Comment by VSCodeTriageBot** [comment]
Hey @DonJayamanne, this issue might need further attention.

@rcgalbo, you can help us out by closing this issue if the problem no longer exists, or adding more information.

**Comment by VSCodeTriageBot** [comment]
This issue has been closed automatically because it needs more information and has not had recent activity. See also our [issue reporting](https://aka.ms/vscodeissuereporting) guidelines.

Happy Coding!

**Comment by DonJayamanne** [comment]
Sorry for not betting back for a while.

> . In a vanilla text document, when I hover over the function name I am able to view the docstring.

Does this also work with plain python files inside the `futures` folder?

* What is the folder structure
Can you provide a very simple repo or folder structure with files/contents to replicate this issue
Instructions such as 
    * Or just create a github repo or a folder and zip it up and upload here (without the Python env and without any external packages)
    * E.g. create a folder named `mypackage` and add `__init__.py` in there...
    * Then create a folder named `mynotebooks` and add the ipynb in there, etc...

**Comment by VSCodeTriageBot** [comment]
This issue has been closed automatically because it needs more information and has not had recent activity. See also our [issue reporting](https://aka.ms/vscodeissuereporting) guidelines.

Happy Coding!
